lib/main.dart

```diff
@@ -1,9 +1,33 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:google_fonts/google_fonts.dart';
 import 'package:instreal/l10n/index.dart';
 import 'package:instreal/presentation/screens/home/home.dart';
 
-void main() {
+import 'firebase_options.dart';
+
+void main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp(
+    options: DefaultFirebaseOptions.currentPlatform,
+  );
+
+  if (kDebugMode) {
+    try {
+      FirebaseFirestore.instance.useFirestoreEmulator(
+        defaultTargetPlatform == TargetPlatform.android
+            ? "10.0.2.2"
+            : "localhost",
+        8080,
+      );
+    } catch (e) {
+      // ignore: avoid_print
```

Comment:
there's a function i think called debug stack print or something, which you don't need to avoid the line.

Reply:
`debugPrintStack` or `debugPrint` doesn't take `Error`. Technically we don't even have `avoid_print` rule in the analyzer config so I can remove it.
Anyway, the codes here is only for debug build so I don't mind just leaving it as it is.

Reply:
I think we have it in the analyzer by default, but you can dismiss it. Anyway, I do agree with Dart that we shouldn't put print :)